From 151ab9128ec3fade3aa814d4003f7dd085ca7113 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Tue, 7 May 2013 15:57:45 +0900
Subject: [PATCH 9/9] Revert "ext/dfbvideosink: workaround: recreate a surface
 after changing the pixel format"

This reverts commit 65a1f30e53b18f575897df16329d35c3081fde1b.

The bug that occurs when changing the pixel format has already been
repaired in DirectFB, fixed at "f07f29ddf8889d166e748c6a7bf48e788ea46eca".
---
 ext/directfb/dfbvideosink.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/ext/directfb/dfbvideosink.c b/ext/directfb/dfbvideosink.c
index e7cd7e4..f2558e3 100644
--- a/ext/directfb/dfbvideosink.c
+++ b/ext/directfb/dfbvideosink.c
@@ -2046,13 +2046,6 @@ gst_dfbvideosink_setcaps (GstBaseSink * bsink, GstCaps * caps)
       gst_dfbvideosink_setup_meram (dfbvideosink, caps, video_width,
           video_height);
 #endif /* defined(HAVE_SHMERAM) */
-
-      /* workaround: recreate the surface after changing the color. */
-      if (dfbvideosink->primary) {
-        dfbvideosink->primary->Release (dfbvideosink->primary);
-        dfbvideosink->layer->GetSurface (dfbvideosink->layer,
-            &dfbvideosink->primary);
-      }
     }
   }
 #if defined(HAVE_SHVIO)
-- 
1.8.1.2

